### PR TITLE
[DO NOT MERGE] COMMERCE-11276 Price on Application on Price Lists & Promotions

### PR DESCRIPTION
**Pull request by jaydawu**
https://issues.liferay.com/browse/COMMERCE-11276
Resent from https://github.com/liferay-commerce/liferay-portal/pull/3841

Add automated tests for internal review under the test plan https://issues.liferay.com/browse/COMMERCE-11682

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase [hunk 1]
```diff
@@ -434,6 +434,73 @@ definition {
...
+			}
+		}
+
+		task ("Then the buyer can't go directly with the checkout, a quote must be submitted") {
+			CommerceNavigator.gotoMiniCart();
+
+			AssertElementPresent(
+				key_text = "Request a Quote",
+				locator1 = "Button#ANY");
+
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase:490]: Can you please add a check to ensure that the checkout button is not able to be clicked?
Thanks!

**Review by pietrobocale — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase [hunk 2]
```diff
@@ -572,6 +639,362 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-11683. Buyer can use Price on Application with default setting"
+	@priority = 5
+	test CanUsePriceOnApplication {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase:644]: Can you change the name to CanUsePriceOnApplicationOnPriceList the one that was present in the ticket is wrong

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase [hunk 3]
```diff
@@ -572,6 +639,362 @@ definition {
...
+		task ("Then the product is not marked as Price on Application") {
+			AssertElementNotPresent(
+				key_productName = "U-Joint",
+				locator1 = "CommerceAccelerators#PRODUCT_CARD_PRICE_ON_APPLICATION");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-11685. Buyer can use Price on Application with eligible channel"
+	@priority = 5
+	test CanUsePriceOnApplicationChannelEligibility {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase:831]: Can you change the name in: "CanPriceListUsePriceOnApplicationChannelEligibility"
Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase [hunk 4]
```diff
@@ -572,6 +639,362 @@ definition {
...
+		task ("Then the product is not marked as Price on Application") {
+			AssertElementNotPresent(
+				key_productName = "U-Joint",
+				locator1 = "CommerceAccelerators#PRODUCT_CARD_PRICE_ON_APPLICATION");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-11686. Buyer can use Price on Application with eligible order type"
+	@priority = 5
+	test CanUsePriceOnApplicationOrderType {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase:917]: CanPriceListUsePriceOnApplicationOrderType

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase [hunk 5]
```diff
@@ -572,6 +639,362 @@ definition {
...
+
+		task ("Then the product is marked as Price on Application") {
+			AssertElementPresent(
+				key_productName = "U-Joint",
+				locator1 = "CommerceAccelerators#PRODUCT_CARD_PRICE_ON_APPLICATION");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-11684. Buyer can use Price on Application with eligible account"
+	@priority = 5
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommercePriceOnApplication.testcase:684]: Can you change the priority to "4"

Thanks!

**Review by pietrobocale — CHANGES_REQUESTED** [review]

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by pietrobocale** [comment]
New PR: #3882

**Comment by pietrobocale** [comment]
ci:close

**Pull request closed without merging**